grpc-interface-support/src/main.py

```diff
@@ -25,14 +26,102 @@
 from velocitas_lib import (
     get_programming_language,
     get_project_cache_dir,
+    get_workspace_dir,
     obtain_local_file_path,
 )
 from velocitas_lib.functional_interface import get_interfaces_for_type
-from velocitas_lib.text_utils import create_truncated_string
 
 DEPENDENCY_TYPE_KEY = "grpc-interface"
 
 
+def extract_zip(file_path: str, extract_to: str) -> str:
+    """Extract a zip file.
+
+    Args:
+        file_path (str): The file path to the zip.
+        extract_to (str): The file path to extract to.
+
+    Returns:
+        str: The file path to the extracted folder.
+    """
+    with zipfile.ZipFile(file_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
+        return extract_to
+
+
+def discover_proto_files_in_directory(
+    directory: str, root: Optional[str] = None
+) -> List[proto.ProtoFileHandle]:
+    """
+    Recursively search for .proto files under the specified directory.
+
+    Args:
+        directory (str): The path to the directory to search in.
+        root (Optional[str]): The optional root for zip directories.
+
+    Returns:
+        List[proto.ProtoFileHandle]: A list of file paths, relative to the search directory, each pointing to a proto file.
+    """
+    proto_files = []
+    if root is not None:
+        directory = os.path.join(directory, root)
```

Comment:
shouldn't this be the other way round? root normally means the base of the path
```suggestion
        directory = os.path.join(root, directory)
```